### ISSUE TITLE
Exception when closing RAFCON and keeping state machine running

### ISSUE THREAD

**Issue by franzlst** [DLR-RM/RAFCON#636]
Steps to reproduce:
1. Create new state machine
2. Run it
3. Close RAFCON
4. "Keep running"

Error: `If a state machine is started and closed while execution and allowed to stop it and close this cause widgets in the state editor to fail with a lot of tracebacks.`

What is happening?

The GUI is destroyed, causing all state machines to be deleted. The `delete_all_state_machines` method sets the `active_state_machine_id` of the `StateMachineManager` to `None`. 

The setter of `active_state_machine_id` raises the mentioned exception, as the following condition is not met: `if not core_singletons.state_machine_execution_engine.finished_or_stopped() and state_machine_id != self._active_state_machine_id:`

A state machine is running, therefore the activate state machine id cannot be set to `None`. The error makes sense. The question is:

Do we want to allow a state machine to be deleted when it is still running? (Intuitively, I would say no)

Depending on the answer, we either have to adjust the condition to allow the activate id to be set to `None` in any case (possible side effects) or skip the running state machine and only destroy the other ones.

Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2018-12-05 10:45:23+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
I moved it to the needed task column in the project because that needs to be resolved before doing an release. It is needed because the traceback shows clearly that the instance trys to destroy respective state machine. Further I will open another issue because this is not the one I mentioned in the project note that was transformed to this issue.

Originally created by @Rbelder  at 2018-12-05 12:37:14+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Maybe we should create a test to cover this.

Originally created by @Rbelder  at 2018-12-05 13:07:39+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
In my opinion all state machine models should be deleted and all state machines except of the running one.
I can care about this.

Originally created by @sebastian-brunner  (Sebastian.Brunner@dlr.de) at 2018-12-06 08:53:57+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
I also think that the models if possible should be destroyed. The seen traceback yesterday was showing that it is tried to destroy all state machines on the core level. So that we do talk about the same thing here the traceback:
```
2018-12-06 11:43:44:    ERROR - rafcon.gui.models.state_machine_manager:  The result type is <type 'exceptions.AttributeError'> and the full notification 	'after': True
	'model': <rafcon.gui.models.state_machine_manager.StateMachineManagerModel object at 0x7f396b54de50>
	'prop_name': state_machine_manager
	'instance': <rafcon.core.state_machine_manager.StateMachineManager object at 0x7f396b68fb50>
	'method_name': active_state_machine_id
	'args': (<rafcon.core.state_machine_manager.StateMachineManager object at 0x7f396b68fb50>, None)
	'kwargs': {}
	'result': Active state machine can not be changed because state machine execution is active.
Traceback (most recent call last):
  File "/net/notos/home_local/dark_room/in_progress/rafcon/foreign_packages/repos/gtkmvc3/gtkmvco/gtkmvc3/observable.py", line 47, in wrapper
    res = _func(*args, **kwargs)
  File "/net/notos/home_local/dark_room/taskcontrol/source/rafcon/core/state_machine_manager.py", line 177, in active_state_machine_id
    raise AttributeError("Active state machine can not be changed because state machine execution is active.")
AttributeError: Active state machine can not be changed because state machine execution is active.
Traceback (most recent call last):
  File "/net/notos/home_local/dark_room/taskcontrol/source/rafcon/gui/controllers/menu_bar.py", line 460, in on_quit_activate
    self.on_delete_check_sm_running()
  File "/net/notos/home_local/dark_room/taskcontrol/source/rafcon/gui/controllers/menu_bar.py", line 523, in on_delete_check_sm_running
    gui_singletons.main_window_controller.prepare_destruction()
  File "/net/notos/home_local/dark_room/taskcontrol/source/rafcon/gui/controllers/main_window.py", line 658, in prepare_destruction
    rafcon.core.singleton.state_machine_manager.delete_all_state_machines()
  File "/net/notos/home_local/dark_room/taskcontrol/source/rafcon/core/state_machine_manager.py", line 51, in delete_all_state_machines
    self.active_state_machine_id = None
  File "/net/notos/home_local/dark_room/in_progress/rafcon/foreign_packages/repos/gtkmvc3/gtkmvco/gtkmvc3/observable.py", line 47, in wrapper
    res = _func(*args, **kwargs)
  File "/net/notos/home_local/dark_room/taskcontrol/source/rafcon/core/state_machine_manager.py", line 177, in active_state_machine_id
    raise AttributeError("Active state machine can not be changed because state machine execution is active.")
AttributeError: Active state machine can not be changed because state machine execution is active.
```

Originally created by @Rbelder  at 2018-12-06 10:45:24+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Fixed with pull request #642

Originally created by @sebastian-brunner  (Sebastian.Brunner@dlr.de) at 2018-12-12 07:49:13+00:00 (moved from RMC internal repository)
